games-fps/urbanterror/urbanterror-4.3.4_p20211208.ebuild

```diff
@@ -57,12 +60,16 @@ pkg_pretend() {
 }
 
 src_configure() {
-	default
-
 	tc-export CC
+
+	default
 }
 
 src_compile() {
+	# -fstrict-aliasing breaks network connections to any server.
+	# Upstream bug is https://github.com/mickael9/ioq3/issues/40.
+	append-flags "-fno-strict-aliasing"
```

Comment:
It appears that upstream is already disabling strict aliasing: https://github.com/mickael9/ioq3/blob/c988f7669bd70b59beb384563d4c8304062ae011/Makefile#L321

Could it be that you ran into this on a system which had explicitly enabled this (which nobody should do)?

Reply:
Sure, as far as I know we allow to set custom CFLAGS etc. in Gentoo for nearly every package and the old patch I carried from the previous maintainer hasufell strips the defaults off and replaces them with a variable set by portage (or the toolchain) of the build host. Actually I forgot to search for that in upstream's Makefile, so my bug at upstream is most probably invalid, thanks for pointing into the right direction there.
I'd stick with this approach and try to remember grepping the Makefiles for filtered flags next time.

Reply:
I wonder if we should use `filter-flag -fstrict-aliasing` instead.

Reply:
Uh, I just discovered that we remove `-fno-strict-aliasing` with a custom gentoo patch: https://github.com/gentoo/gentoo/blob/770a83e33e69c614e533028f9c85b7165ed1f077/games-fps/urbanterror/files/urbanterror-4.3.4_p20180708-fix-build_system.patch#L16-L18

I am pretty sure that we shouldn't do that.

Reply:
Really... kept hearing about this package and strict-aliasing in #gentoo-games but didn't think it was because it was straight up removed :eyes: Yes, I'd change the patch to keep it personally. Warnings flags can be left alone as well, this isn't -Werror.

It's good to cleanup some overaggressive flags, but not flags that are necessary to function properly (even if it's not good practice to rely on -fno-strict-aliasing)